src/main/scala/is/hail/utils/richUtils/RichRDD.scala

```diff
@@ -184,18 +184,24 @@ class RichRDD[T](val r: RDD[T]) extends AnyVal {
       .subsetPartitions((0 to idxLast).toArray)
   }
   
-  def writePartitions(path: String, write: (Int, Iterator[T], OutputStream) => Long): Array[Long] = {
+  def writePartitions(path: String,
+    write: (Int, Iterator[T], OutputStream) => Long,
+    mapPartitionIDs: Option[Array[Int]] = None,
+    nPartsForString: Option[Int] = None): Array[Long] = {
```

Comment:
Does it ever make sense to pass `None` and `Some(...)`? It seems like it should be `remapPartitions: Option[(Array[Int], Option[Int])]`:

```scala
val (nPartitions, remap) = remapPartitions match {
  None => (r.getNumPartitions, x => x)
  Some(remap, None) => (r.getNumPartitons, remap.apply _)
  Some(remap, Some(n)) => (n, remap.apply _) 
}
```

Reply:
Good call, and I actually think only the first and third cases are relevant to I changes it to:
`remapPartitions: Option[(Array[Int], Int)]`